indico/modules/events/api.py

```diff
@@ -147,7 +150,248 @@ def export_event(self, aw):
         return expInt.event(self._idList)
 
 
-class CategoryEventFetcher(IteratedDataFetcher):
+class SerializerBase(object):
+    """Common methods for different serializers"""
+
+    def _serialize_access_list(self, obj):
+        data = {'users': [], 'groups': []}
+        for manager in obj.get_access_list():
+            if manager.principal_type in (PrincipalType.user, PrincipalType.email):
+                data['users'].append(manager.email)
+            elif manager.principal_type == PrincipalType.multipass_group:
+                data['groups'].append(manager.name)
+        return data
+
+    def _serialize_date(self, date):
+        if date:
+            date = date.astimezone(self._hook._tz)
+            return {
+                'date': str(date.date()),
+                'time': str(date.time()),
+                'tz': str(date.tzinfo)
+            }
+
+    def _serialize_person(self, person, person_type, can_manage=False):
+        """ Method to serialize an event associated person (author, co-author,
+        speaker, etc).
+        """
+
+        if person:
+            data = {
+                '_type': person_type,
+                '_fossil': self.fossils_mapping['person'].get(person_type, None),
+                'fullName': person.get_full_name(last_name_upper=False, abbrev_first_name=False),
+                'id': unicode(person.id),
+                'affiliation': person.affiliation,
+                'emailHash': md5(person.email).hexdigest() if person.email else None
```

Comment:
I remember suggesting to add `first_name` and `last_name` in addition to the fullName